MTES-MCT/ecobalyse
# feat: add components to objects

## :wrench: Problem

We want to be able to model the component notion. See [the card](https://www.notion.so/Objet_V0-S-lection-des-composants-79ca63b368524ee0b1f06321aa1878c5) for more details.

## :cake: Solution

Add a level above the processes with a name and a quantity.


## :rotating_light:  Points to watch/comments

Currently, components are built on the file from the example file. Their uniqueness is based on their name, it should be improved later on with component ids and a file listing the possible components.

It's not possible to modify processes on purpose, it should come later on.

Tests need to be added.

## :desert_island: How to test

- Go on the object tab
- Select the "Chaise", score should be 422
- Select the  "Table", score should be 3 979
- Add and remove some components: score should be updated accordingly
- Expand the "Procédés" summary and check that the totals impacts and mass of the component is the impacts and mass of the process multiplied by the quantity of the component like below

![image](https://github.com/user-attachments/assets/63069916-bf8d-463e-910d-9ee27004c582)
- Check that the bookmarks work as expected
